src/main/java/nl/tudelft/scrumbledore/userinterface/GameDisplay.java

```diff
@@ -275,21 +272,25 @@ private static void playerStatus() {
   private static void levelStatus() {
     if (currentGame.getCurrentLevel().getNPCs().isEmpty()
         && currentGame.getCurrentLevel().getEnemyBubbles().isEmpty()) {
-      if (currentGame.remainingLevels() == 0) {
-        Logger.getInstance().log("Player completed the game successfully.");
-
-        animationTimer.stop();
-
-        winDialog();
-      } else {
-        Logger.getInstance().log("Player advanced to the next level.");
-        staticContext.setFill(Color.WHITE);
-        staticContext.fillText(Constants.ADVANCINGLABEL, (Constants.LEVELX / 2) - 100,
-            (Constants.LEVELY / 2) - 130);
-
-        currentGame.goToNextLevel();
-        GameDisplay.renderStatic();
+      if (endTimer <= 0) {
+        if (currentGame.remainingLevels() == 0) {
+          Logger.getInstance().log("Player completed the game successfully.");
+  
+          animationTimer.stop();
+  
+          winDialog();
+        } else {
+          Logger.getInstance().log("Player advanced to the next level.");
+          staticContext.setFill(Color.WHITE);
+          staticContext.fillText(Constants.ADVANCINGLABEL, (Constants.LEVELX / 2) - 100,
+              (Constants.LEVELY / 2) - 130);
+  
+          currentGame.goToNextLevel();
+          GameDisplay.renderStatic();
+        }
+        endTimer = Constants.REFRESH_RATE * 4;
       }
+      endTimer--;
```

Comment:
This is not correct yet: the endtimer should be decremented by delta instead of 1 each cycle. I thibk you should take a snapshot of game.getSteps() when the player finishes the level, and then compare the snapshot to the current game.getSteps() each cycle. Advance to the level if the differencd between these is larger than 4 times the refresh_rate.